lib/onelogin/ruby-saml/response.rb

```diff
@@ -596,6 +629,40 @@ def validate_signature
         true
       end
 
+      # Validates the Signature using idp_cert_multi and idp_cert_fingerprint_multi
+      # @return [Boolean] True if not contains a Signature or if the Signature is valid, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate_signature_multi_cert
+        fingerprint_multi = settings.get_fingerprint_multi
+        idp_cert_multi = settings.get_idp_cert_multi
+
+        if fingerprint_multi.empty? && !validate_fingerprint(fingerprint_multi, idp_cert_multi)
+          error_msg = "Invalid Signature on SAML Response"
+          return append_error(error_msg)
+        end
+
+        true
+      end
+
+      def validate_fingerprint(fingerprint_multi, idp_cert_multi)
+        response_signed = REXML::XPath.first(
+          document,
+          "/p:Response[@ID=$id]",
+          { "p" => PROTOCOL, "ds" => DSIG },
+          { 'id' => document.signed_element_id }
+        )
+        doc = (response_signed || decrypted_document.nil?) ? document : decrypted_document
+
+        fingerprint_multi.zip(idp_cert_multi).any? do |fingerprint, idp_cert|
+          opts = {}
+          opts[:fingerprint_alg] = settings.idp_cert_fingerprint_algorithm
+          opts[:cert] = idp_cert
```

Comment:
This is kind of ugly, but I think it's probably better we stick closer to the logic in `#validate_signature`